### ISSUE TITLE
Changing color of roador enhancing contrast and image compression

### ISSUE THREAD

**Issue by smajser** [openstreetcam/openstreetview.org#142]
I am using map a lot, and I find it very helpful but there is one problem. Contrast between background and streets is very weak and it is quite difficult to see streets clearly.

Is it possible to add feature to enhance contrast or change color of streets by user preferences? 

Also it would be good if there is an option to compress images taken by "JPEG recording" before uploading them. A lot of time and bandwidth could be saved with this.

**Comment by alexandrui-telenav** [comment]
Hi @smajser , are you talking about the map in the app or the website?

**Comment by smajser** [comment]
Hi Alex,

I am talking about map in app.

G: smajser1
S: uros.racic
P: +381 63 45-48-45
L: https://www.linkedin.com/in/uros-racic-0a9555120/
T: @racic_uros


------ Original Message ------

**Comment by alexandrui-telenav** [comment]
It's a good point, we're working on a map style that increases the contrast. 

On the JPEG encoding suggestion, the reasoning for letting the JPEG mode in there was that on some devices the video encoding is not working well due to OS inconsistencies. Moving that to a later point will still lead to the same problems.. So what's the reason you're recording in JPEG mode in the first place, did you have problems with the default?

**Comment by smajser** [comment]
No, I did not have any problems with both modes, all work fine. But when 
I use JPEG my phone takes more pictures.
If I could use JPEG mode and then convert to comppressed format, that 
would benefit in two ways (on my side). First it would reduce upload 
time (I have 1 MB upload), and second the path would be more detailed. 
For the second part, I do not find that it is a problem on routes that I 
go regulary (I can take photos 100 times), but there are roads that are 
less frequently traveled and that would benefit from ability to take 
more pictures at once.

G: smajser1
S: uros.racic
P: +381 63 45-48-45
L: https://www.linkedin.com/in/uros-racic-0a9555120/
T: @racic_uros


------ Original Message ------

**Comment by alexandrui-telenav** [comment]
Got it. To give you a bit of background, the app doesn't make any difference between JPEG or video modes for the photo trigger, but it's  possible that video encoding can't keep up with the recording and then some frames are skipped. 
Doing a post-JPEG encoding involves some technical difficulties ( whole metadata file needs to be parsed and updated ) and you'd also lose more quality ( both JPEG and h264 are lossy formats, hence double quality loss ). 
What we think is the better approach here is to optimise the encoding using the native Android API instead of ffmpeg, there's a good chance that recording with video compression will work without skipped frames afterwards. 
I know this is usually not a solution, but a better Android phone or an iPhone solves this as well as it wouldn't skip frames. Can you share what phone do you use?

**Comment by smajser** [comment]
Well, I did not think that way, but I understand that part.
My phone is Sony Xperia XA



G: smajser1
S: uros.racic
P: +381 63 45-48-45
L: https://www.linkedin.com/in/uros-racic-0a9555120/
T: @racic_uros


------ Original Message ------
